### DISCUSSION TITLE
May I know the sonar.properties file configuration to work with this plugin

### DISCUSSION THREAD

**Discussion by govindappaarun** [koliveira15/nx-sonarqube#56] [General]
Hi @koliveira15 , 
I am trying to migrate an existing monorepo with sonar to run with your plugin to have it cached with Nx. 
May I know the min config thats required in sonar.properties, we are facing issues with indexing, "cant index a file twice " when we use our  using existing sonar.properties

**Comment by koliveira15** [comment]
Hi @govindappaarun,

The minimum config is defined by [Sonar's Analysis Parameters](https://docs.sonarsource.com/sonarqube/latest/analyzing-source-code/analysis-parameters/). From a little digging, its probably because you have an exclusion in your config while also using the Nx plugin which creates that config on the fly. I would recommend removing the sonar config properties file completely and you can also [log the sonar properties ](https://github.com/koliveira15/nx-sonarqube/blob/main/packages/nx-sonarqube/src/executors/scan/utils/utils.ts#L161C7-L161C21) from the plugin and compare. Also if you have a repo to reproduce I can take a look as well.

**Reply by govindappaarun** [reply]
@koliveira15  May I know for a monorepo ( pacakge based) SONAR configuration  for a project with more than 10 packages and using SonarQube for reporting, any sample repo which is using this plugin

**Reply by govindappaarun** [reply]
I am asking this question to see if I have any miss configuration, I used this plugin and tried to execute sonar scan for each of my 50 modules and then scan of each of the packages is taking 5mins nearly to analyze and they have to be executed in sequential. Which is more worse than the current time ( without nx) to scan all the 50 modules is around 15mins
